src/lib/viewers/archive/ArchiveExplorer.js

```diff
@@ -132,19 +132,65 @@ class ArchiveExplorer extends React.Component {
      */
     handleClickFullPath = fullPath => this.setState({ fullPath });
 
+    /**
+     * Handle search input, update view state
+     *
+     * @param {string} query - raw query string in the search bar
+     * @return {void}
+     */
+    search = query => {
+        const trimmedQuery = query.trim();
+
+        if (!query) {
+            this.setState({
+                searchQuery: query,
+                view: VIEW_FOLDER,
+            });
+            return;
+        }
+
+        if (!trimmedQuery) {
+            this.setState({
+                searchQuery: query,
+            });
+            return;
+        }
+
+        this.setState({
+            searchQuery: query,
+            view: VIEW_SEARCH,
+        });
+    };
+
+    /**
+     * Filter item collection for search query
+     *
+     * @param {Array<Object>} itemCollection - raw data
+     * @param {string} searchQuery - user input
+     * @return {Array<Object>} filtered items for search query
+     */
+    getSearchResult = (itemCollection, searchQuery) => {
+        const trimmedQuery = searchQuery.trim();
```

Comment:
It doesn't seem like this variable is needed, since `.trim()` can be called inline.

Reply:
> It doesn't seem like this variable is needed, since `.trim()` can be called inline.

But the next line is `filter` function, it's like a for loop and the `.trim()` will be called many times.

Reply:
True. I'd consider that a micro-optimization given the amount of logic being executed within the `fuzzysearch` function, but it's fine as-is.